src/paste-markdown-link.ts

```diff
@@ -0,0 +1,52 @@
+import {insertText} from './text'
+
+export function install(el: HTMLElement): void {
+  el.addEventListener('paste', onPaste)
+}
+
+export function uninstall(el: HTMLElement): void {
+  el.removeEventListener('paste', onPaste)
+}
+
+function onPaste(event: ClipboardEvent) {
+  const transfer = event.clipboardData
+  if (!transfer || !hasPlainText(transfer)) return
+
+  const field = event.currentTarget
+  if (!(field instanceof HTMLTextAreaElement)) return
+
+  const text = transfer.getData('text/plain')
+  if (!text) return
+
+  if (isWithinLink(field)) return
+
+  event.stopPropagation()
+  event.preventDefault()
+
+  const selectedText = field.value.substring(field.selectionStart, field.selectionEnd)
+
+  insertText(field, linkify(selectedText, text), {addNewline: false})
+}
+
+function hasPlainText(transfer: DataTransfer): boolean {
+  return Array.from(transfer.types).includes('text/plain')
+}
+
+function isWithinLink(textarea: HTMLTextAreaElement): boolean {
+  const selectionStart = textarea.selectionStart || 0
+
+  if (selectionStart > 1) {
+    const previousChars = textarea.value.substring(selectionStart - 2, selectionStart)
+    return previousChars === ']('
```

Comment:
What do you think about checking for HTML links as well? Since markdown generally supports HTML as well, should we check if we're in a `<a>` tag?

Reply:
I've thought about some other edge cases as well, for example let's say you have this issue template:

```
Repository: [URL of repository]
```

If you select `[URL of repository]` and paste an URL on top of it, with the new feature it would now convert it to `[[URL of repository]](url)`.

Though, I'm not sure we should account for all edge cases or if users will learn quickly that they need to delete the selected text first and then paste the URL in order to have the URL as is.

Regarding detecting if the selected text is within an `<a>` tag:
Do you think of the use case where a user would have an <a>-link in their Markdown, such as `<a href="https://some-url">Docs</a>`, and where they would want to change the `href` by selecting the `https://some-url` part and replace it with another URL?